src/Cake.SqlServer/Backup/RestoreSqlBackupImpl.cs

```diff
@@ -11,65 +12,107 @@ internal static class RestoreSqlBackupImpl
     {
         // if database name is not provided, dbname from the backup is used.
         // if newStoragePath is not provided, system defaults are used
-        internal static void RestoreSqlBackup(ICakeContext context, String connectionString, FilePath backupFile, RestoreSqlBackupSettings settings)
+        internal static void RestoreSqlBackup(ICakeContext context, String connectionString, RestoreSqlBackupSettings settings, IList<FilePath> backupFiles, IList<FilePath> differentialBackupFiles = null)
         {
             using (var connection = SqlServerAliasesImpl.OpenSqlConnection(context, connectionString))
             {
-                var oldDbName = GetDatabaseName(backupFile, connection);
-                var newDatabaseName = settings.NewDatabaseName ?? oldDbName;
-                context.Log.Information($"Using database name '{newDatabaseName}' to be a name for the restored database");
+                bool hasDifferentialBackup = differentialBackupFiles != null && differentialBackupFiles.Any();
+                var mainCommand = GetSqlCommand(context, connection, settings, backupFiles, settings.BackupSetFile, hasDifferentialBackup);
+                mainCommand.ExecuteNonQuery();
+                if (hasDifferentialBackup)
+                {
+                    var differentialCommand = GetSqlCommand(context, connection, settings, differentialBackupFiles, settings.DifferentialBackupSetFile, false, false);
```

Comment:
It would help the readability to name the optional parameters.

Reply:
See my remark above -> I changed the internal design.